src/core/Akka/Actor/ActorTelemetry.cs

```diff
@@ -0,0 +1,104 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ActorTelemetry.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using Akka.Event;
+
+namespace Akka.Actor
+{
+    /// <summary>
+    /// A set of events designed to provide some basic telemetry functions for monitoring actor lifecycles.
+    ///
+    /// We want to track actor starts, stops, and restarts. More detailed metrics, such as mailbox size or message
+    /// processing rates will require something like Phobos [https://phobos.petabridge.com/].
+    /// </summary>
+    /// <remarks>
+    /// Not intended to be sent across network boundaries - should only be processed locally via the <see cref="EventStream"/>.
+    /// </remarks>
+    public interface IActorTelemetryEvent : INoSerializationVerificationNeeded, INotInfluenceReceiveTimeout
+    {
+        /// <summary>
+        /// The actor who emitted this event.
+        /// </summary>
+        IActorRef Subject {get;}
+        
+        /// <summary>
+        /// The implementation type for this actor.
+        /// </summary>
+        Type ActorType { get; }
+    }
+    
+    // Create ActorTelemetryEvent messages for the following events: starting an actor, stopping an actor, restarting an actor
+    public readonly struct ActorStarted : IActorTelemetryEvent
```

Comment:
Is it worth using Structs here if they are getting boxed anyway?

Reply:
I'll benchmark this and find it - what I'd _like_ to do is use structs and get rid of the boxing but that'll require a non-binary-compatible change to the `EventStream` class.

Reply:
So that change would probably be an Akka.NET v1.5 thing.